components/kyma-environment-broker/internal/process/provisioning/initialisation.go

```diff
@@ -73,15 +73,27 @@ func (s *InitialisationStep) initializeRuntimeInputRequest(operation internal.Pr
 		return s.operationManager.OperationFailed(operation, "invalid operation provisioning parameters")
 	}
 
+	var kymaVersion string
+	if pp.Parameters.KymaVersion == "" {
+		log.Info("input builder setting up to work with default Kyma version")
+	} else {
+		kymaVersion = pp.Parameters.KymaVersion
+		log.Infof("setting up input builder to work with %s Kyma version", kymaVersion)
+	}
+
 	log.Infof("create input creator for %q plan ID", pp.PlanID)
-	creator, found := s.inputBuilder.ForPlan(pp.PlanID)
-	if !found {
-		log.Error("input creator does not exist")
+	creator, err := s.inputBuilder.ForPlan(pp.PlanID, kymaVersion)
+	switch {
+	case err == nil:
+		operation.InputCreator = creator
+		return operation, 0, nil
+	case kebError.IsTemporaryError(err):
+		log.Errorf("cannot create temporary input creator for plan %s and version %s: %s", pp.PlanID, kymaVersion, err)
```

Comment:
> "cannot create temporary input creator (..)"

please rephrase it as this "temporary" verb in this context is quite misleading :)